scripts/symbol_to_entrez_id.py

```diff
@@ -0,0 +1,227 @@
+"""
+Module to perform fuzzy mapping of symbols to Entrez IDs.
+
+Uses MyGene for search of gene symbol aliases:
+http://docs.mygene.info/projects/mygene-py/en/latest/
+
+"""
+import time
+import mygene
+import numpy as np
+import pandas as pd
+
+def query_to_map(df_query, target_name, map_to_lists=False):
+    """Convert results of a MyGene query to a dict."""
+    query_map = {}
+    if 'notfound' not in df_query:
+        df_query['notfound'] = np.nan
+    for query, row in df_query[df_query['notfound'].isnull()].iterrows():
+        if map_to_lists:
+            if query in query_map:
+                query_map[query].append(row[target_name])
+            else:
+                query_map[query] = [row[target_name]]
+        else:
+            query_map[query] = row[target_name]
+    return query_map
+
+def invert_map(orig_map):
+    """Invert a dict, possibly containing list values."""
```

Comment:
Rather than possibly, it must contain list value, correct? Wouldn't a single value would cause the following error:

```python
>>> invert_map({'a': ['c123', 'd456'], 'b': 'e789'}
{'c123': 'a', 'd456': 'a', 'e': 'b', '7': 'b', '8': 'b', '9': 'b'}
```

Reply:
I actually ran this function with Michael's example and got  {'c123': 'a', 'd456': 'a'}. I'd worry about single values getting dropped in this implementation.

Reply:
Good catch. I updated the comment to make it clear that all the values must be lists. For my application this should be fine - I don't expect it to be used for any dicts with single values.